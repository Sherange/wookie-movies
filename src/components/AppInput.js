import React from 'react';
import PropTypes from 'prop-types';
import {View, StyleSheet} from 'react-native';
import {Input} from 'react-native-elements';
import { primaryTextColor } from '../constants/theme';

const AppInput = props => {
  const {placeholder} = props;
  return (
    <View style={styles.buttonContainer}>
      <Input placeholder={placeholder} inputStyle={styles.inputStyle} />
    </View>
  );
};

AppInput.propTypes = {
  placeholder: PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  inputContainer: {
    flexDirection: 'row',
    marginVertical: 8,
  },
  inputStyle : {
    color : primaryTextColor
  }
});

export default AppInput;
