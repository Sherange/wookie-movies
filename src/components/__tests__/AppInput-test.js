import 'react-native';
import React from 'react';
import AppInput from '../AppInput';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  renderer.create(<AppInput placeholder={'Search title'} />);
});

test('AppInput snapshot', () => {
  const snapshot = renderer
    .create(<AppInput placeholder={'Search title'} />)
    .toJSON();
  expect(snapshot).toMatchSnapshot();
});
