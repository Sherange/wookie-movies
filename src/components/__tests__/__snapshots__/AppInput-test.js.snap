// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppInput snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        editable={true}
        placeholder="Search title"
        placeholderTextColor="#86939e"
        style={
          Object {
            "color": "#FFFFFF",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ff190c",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
</View>
`;
